composer/core/time.py

```diff
@@ -45,11 +45,11 @@ class TimeUnit(StringEnum):
 
 
 # regex for parsing integers / decimals / scientific notation
-_NUM_REGEX = r'-?[\d.]+(?:e-?\d+)?'
+_NUM_REGEX = r'.+'
 
 # regex for parsing a time string.
-_TIME_STR_REGEX = re.compile(r'^(?:' + r'|'.join(fr'(?:({_NUM_REGEX})({time_unit.value}))' for time_unit in TimeUnit) +
-                             r')$',
+_TIME_STR_REGEX = re.compile(r'^(' + fr'{_NUM_REGEX}' + ')(' +
+                             r'|'.join([fr'{time_unit.value}' for time_unit in TimeUnit]) + r')$',
```

Comment:
Can you please update the comments to be a bit more clear here?

It looks like the `_NUM_REGEX` is adjusted to match everything besides the time_unit. In that case, can we just get rid of that value and in-line it into the `_TIME_STR_REGEX` since it's been simplified? 

Can we also update the comment to be clear on how it works? Something like we match the unit and assume the rest is the number?

Reply:
Sure, corrected it.